oak_functions/loader/src/grpc.rs

```diff
@@ -105,21 +94,43 @@ pub async fn create_and_start_grpc_server<F: Future<Output = ()>>(
         async move |request| handle_request(wasm_handler, policy.clone(), request).await;
 
     let additional_info = config_info.encode_to_vec();
+
     // A `Service` is needed for every connection. Here we create a service using the
     // `wasm_handler`.
-    tonic::transport::Server::builder()
-        .add_service(StreamingSessionServer::new(
-            AttestationServer::create(
-                tee_certificate,
-                request_handler,
-                additional_info,
-                ErrorLogger { logger },
-            )
-            .context("Couldn't create remote attestation server")?,
-        ))
-        .serve_with_shutdown(*address, terminate)
-        .await
-        .context("Couldn't start server")?;
+    match request_model {
+        RequestModel::Unary => {
+            tonic::transport::Server::builder()
+                .add_service(
+                    grpc_unary_attestation::proto::unary_session_server::UnarySessionServer::new(
+                        grpc_unary_attestation::server::AttestationServer::create(
+                            tee_certificate,
+                            request_handler,
+                            additional_info,
+                            ErrorLogger { logger },
+                        )
+                        .context("Couldn't create remote attestation server")?,
+                    ),
+                )
+                .serve_with_shutdown(*address, terminate)
+                .await
+                .context("Couldn't start server")?;
```

Comment:
It seems this could be moved out of the match -- in fact, only the actual added service is different, is it possible to just create that differently?

Reply:
The type of the service that is registered, as well as the type of the Router returned by registering is different, so we can't move it out of the match easily.

Reply:
Ah thanks, perhaps worth a comment, it may not be that obvious, but up to you.

Reply:
Added a comment. :)